examples/streamed_sockets/speedtest/receiver.cc

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2019 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#include <deque>
+
+#include "../../../blocks/xterm/progress.h"
+#include "../../../blocks/xterm/vt100.h"
+#include "../../../bricks/dflags/dflags.h"
+#include "../../../bricks/net/tcp/tcp.h"
+#include "../../../bricks/strings/printf.h"
+#include "../../../bricks/time/chrono.h"
+
+DEFINE_uint16(port, 9001, "The port to use.");
+DEFINE_double(receive_buffer_gb, 0.5, "The size of the buffer the read the data into.");
+DEFINE_double(window_size_seconds, 5.0, "The length of sliding window the throughput within which is reported.");
+DEFINE_double(window_size_gb, 20.0, "The maximum amount of data per the sliding window to report the throughput.");
+DEFINE_double(output_frequency, 0.1, "The minimim amount of time, in seconds, between terminal updates.");
+DEFINE_double(max_seconds_of_silence, 1.5, "Terminate the connection if it sends nothing for this long, in seconds.");
+
+int main(int argc, char** argv) {
+  ParseDFlags(&argc, &argv);
+
+  using namespace current::net;
+  using namespace current::vt100;
+
+#ifndef NDEBUG
+  {
+    std::cout << yellow << bold << "warning" << reset << ": unoptimized build";
+#if defined(CURRENT_POSIX) || defined(CURRENT_APPLE)
+    std::cout << ", run " << cyan << "NDEBUG=1 make clean all";
+#endif
+    std::cout << std::endl;
+  }
+#endif
+
+  std::chrono::microseconds const t_window_size(static_cast<int64_t>(FLAGS_window_size_seconds * 1e6));
```

Comment:
As far as I remember, we agreed to keep `const`s on the left hand side?

Reply:
Fixed in `6932cf72`.